pkg/daemon/ceph/client/subvolumegroup.go

```diff
@@ -55,3 +58,71 @@ func DeleteCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterI
 	logger.Infof("successfully deleted cephfs %q subvolume group %q", volName, groupName)
 	return nil
 }
+
+// PinCephFSSubVolumeGroup pin the cephfs subvolume group
+func PinCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterInfo, volName string, cephFilesystemSubVolumeGroup *cephv1.CephFilesystemSubVolumeGroup, cephFilesystemSubVolumeGroupName string) error {
+	logger.Infof("pinning cephfs subvolume group %q of filesystem %q", cephFilesystemSubVolumeGroupName, volName)
+
+	args := []string{"fs", "subvolumegroup", "pin", volName, cephFilesystemSubVolumeGroupName}
+	if cephFilesystemSubVolumeGroup.Spec.Pinning.Distributed != nil {
+		setting := strconv.Itoa(*cephFilesystemSubVolumeGroup.Spec.Pinning.Distributed)
+		args = append(args, "distributed", setting)
+	} else if cephFilesystemSubVolumeGroup.Spec.Pinning.Export != nil {
+		setting := strconv.Itoa(*cephFilesystemSubVolumeGroup.Spec.Pinning.Export)
+		args = append(args, "export", setting)
+	} else if cephFilesystemSubVolumeGroup.Spec.Pinning.Random != nil {
+		setting := strconv.FormatFloat(*cephFilesystemSubVolumeGroup.Spec.Pinning.Random, 'f', -1, 64)
```

Comment:
How many decimals accuracy does this give? We probably don't need more than a few decimals supported

Reply:
It is keeping till 16 digit `4.145654334355354` https://go.dev/play/p/b4dCgmIvaec
 but it's totally upto user what they input